src/main/java/org/alliancegenome/curation_api/model/entities/InformationContentEntity.java

```diff
@@ -23,6 +24,7 @@
 @Data
 @EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
 @ToString(callSuper = true)
+@JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
I get multiple failures in the integration tests without this, e.g.
`com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "crossReferences" (class org.alliancegenome.curation_api.model.entities.InformationContentEntity), not marked as ignorable (9 known properties: "updatedBy", "obsolete", "dateCreated", "dbDateCreated", "dateUpdated", "createdBy", "internal", "dbDateUpdated", "curie"])
 at [Source: (String)"{"entity":{"type":"Gene","createdBy":{"internal":false,"obsolete":false,"dbDateCreated":"2024-01-11T12:58:36.965568Z","dbDateUpdated":"2024-01-11T12:58:36.965638Z","id":200000000,"uniqueId":"Local|Dev User|test@alliancegenome.org","firstName":"Local","lastName":"Dev User"},"updatedBy":{"internal":false,"obsolete":false,"dbDateCreated":"2024-01-11T12:58:36.965568Z","dbDateUpdated":"2024-01-11T12:58:36.965638Z","id":200000000,"uniqueId":"Local|Dev User|test@alliancegenome.org","firstName":"Local","[truncated 8557 chars]; line: 1, column: 3033] (through reference chain: org.alliancegenome.curation_api.response.ObjectResponse["entity"]->org.alliancegenome.curation_api.model.entities.Gene["geneSymbol"]->org.alliancegenome.curation_api.model.entities.slotAnnotations.geneSlotAnnotations.GeneSymbolSlotAnnotation["evidence"]->java.util.ArrayList[0]->org.alliancegenome.curation_api.model.entities.InformationContentEntity["crossReferences"])
	at org.alliancegenome.curation_api.crud.controllers.GeneITCase.editGene(GeneITCase.java:192)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: 
Unrecognized field "crossReferences" (class org.alliancegenome.curation_api.model.entities.InformationContentEntity), not marked as ignorable (9 known properties: "updatedBy", "obsolete", "dateCreated", "dbDateCreated", "dateUpdated", "createdBy", "internal", "dbDateUpdated", "curie"])
 at [Source: (String)"{"entity":{"type":"Gene","createdBy":{"internal":false,"obsolete":false,"dbDateCreated":"2024-01-11T12:58:36.965568Z","dbDateUpdated":"2024-01-11T12:58:36.965638Z","id":200000000,"uniqueId":"Local|Dev User|test@alliancegenome.org","firstName":"Local","lastName":"Dev User"},"updatedBy":{"internal":false,"obsolete":false,"dbDateCreated":"2024-01-11T12:58:36.965568Z","dbDateUpdated":"2024-01-11T12:58:36.965638Z","id":200000000,"uniqueId":"Local|Dev User|test@alliancegenome.org","firstName":"Local","[truncated 8557 chars]; line: 1, column: 3033] (through reference chain: org.alliancegenome.curation_api.response.ObjectResponse["entity"]->org.alliancegenome.curation_api.model.entities.Gene["geneSymbol"]->org.alliancegenome.curation_api.model.entities.slotAnnotations.geneSlotAnnotations.GeneSymbolSlotAnnotation["evidence"]->java.util.ArrayList[0]->org.alliancegenome.curation_api.model.entities.InformationContentEntity["crossReferences"])
	at org.alliancegenome.curation_api.crud.controllers.GeneITCase.editGene(GeneITCase.java:192)`

Reply:
so the api is trying to create a infocontent but what is getting passed in is a reference? there for the unknown fields?

Reply:
i am wondering if we should change the model to represent this? because this is like sending a gene to a biologicalentity it doesn't know about the specific gene field. plus i also maintain that this should work and not fail

Reply:
I think the whole evidence part of the model needs a rethink.  There was some discussion about it a while ago but I forget the conclusion.  I'd be in favour of ditching InformationContentEntity and just using Reference directly, then using additional slots for other evidence types where necessary.

Reply:
is the error happening on the API side or client side when the integration tests run?

Reply:
Could that also explain the other issue I'm seeing with the allele/updateDetail endpoint?  Doing an update with alleleGeneAssociations doesn't delete the associations, yet the integration test that does exactly the same API calls works.

Reply:
API side

Reply:
Ya that could be... I think we could put some code in the initialization of the objectmapper to check that its being created

Reply:
It does seem to be - I turned on the logging message in RestDefaultObjectMapper() and it showed up in the IT logs